pkg/store/etcd.go

```diff
@@ -115,6 +110,30 @@ func (s *Etcd) Put(key string, value []byte) error {
 	return nil
 }
 
+// Create the entire path for a directory that does not exist
+func (s *Etcd) createDirectory(path string) error {
+	if _, err := s.client.CreateDir(normalize(path), 10); err != nil {
+		if etcdError, ok := err.(*etcd.EtcdError); ok {
+			if etcdError.ErrorCode != 105 { // Skip key already exists
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// Periodically renew a key with an Ephemeral flag set
+func (s *Etcd) periodicRenew(key string, value []byte, ttl time.Duration) {
+	renew := time.NewTicker(40 * time.Second)
```

Comment:
Same here:

Make `40` either:
- a `const`
- a `WriteOption`
- a `Config`

This could be named `heartbeat` or something like that (this is exactly what the heartbeat was supposed to be like)